Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/SharedKeyHttpObservable.java

```diff
@@ -74,12 +140,33 @@ public SharedKeyHttpObservable removeContentLength() {
         return this;
     }
 
-    public SharedKeyHttpObservable setContentType(@NotNull String type) {
-        getDefaultHeaderGroup().updateHeader(new BasicHeader("Content-Type", type));
-        return this;
+    @Override
+    public Observable<CloseableHttpResponse> executeReqAndCheckStatus(HttpEntityEnclosingRequestBase req,
+                                                                      int validStatueCode, List<NameValuePair> pairs) {
+        long entityLen = req.getEntity() != null ? req.getEntity().getContentLength() : 0;
+
+        // adls gen2 deployable needs set content-length to generate shared key
+        // but httpclient auto adds this header and calculates length when executing
+        // so remove this header after key generation otherwise header already exists exp happens
+        if (entityLen != 0) {
```

Comment:
I suppose here needn't to check `entityLen` value.

Reply:
Append req needs this to generate key 

---
In reply to: [271166692](https://github.com/Microsoft/azure-tools-for-java/pull/2961#discussion_r271166692) [](ancestors = 271166692)